third_party/pcsc-lite/naclport/server_clients_management/src/client_request_processor.cc

```diff
@@ -410,33 +437,118 @@ GenericRequestResult PcscLiteClientRequestProcessor::SCardConnect(
   tracer.LogEntrance();
 
   LONG return_code = SCARD_S_SUCCESS;
-  if (!s_card_handles_registry_.ContainsContext(s_card_context))
+  if (update_card_handles_registry &&
```

Comment:
Is it correct that we guard this check by `update_card_handles_registry`? I think `SCARDCONTEXT` should always be valid.